src/Events/lib.mo

```diff
@@ -81,43 +80,90 @@ module {
         #Public;
         // Denotes an event with limited access.
         #Private : Allowlist;
+        // Denotes an event with limited access to holders of a certain NFT.
+        #Holders : HolderAccess;
+    };
+
+    public type BalanceInterface = actor {
+        // Returns the amount of NFTs of the given user.
+        launchpadBalanceOf : query (user : Principal) -> async Nat;
```

Comment:
I think my primary critique of this pr would be that it requires additional lp specific interface, instead of composing with existing standards.

Noteworthy that if we were on dip, we could simply use balanceOf, which would moot my point.

Reply:
This is exactly why I specified a new interface, so we do not force a specific one on the NFT canister.

Reply:
Makes some sense for sure